homeassistant/components/amberelectric/sensor.py

```diff
@@ -0,0 +1,591 @@
+
+                if meta.range is not None:
+                    datum["range_min"] = meta.range.min
+                    datum["range_max"] = meta.range.max
+
+                data["forecasts"].append(datum)
+
+        data[ATTR_ATTRIBUTION] = ATTRIBUTION
+        return data
+
+
+class AmberPriceSpikeSensor(CoordinatorEntity, SensorEntity):
+    """Amber Price Spike Sensor."""
+
+    def __init__(
+        self,
+        platform_name: str,
+        site_id: str,
+        data_service: AmberDataService,
+        coordinator: DataUpdateCoordinator,
+    ) -> None:
+        """Initialize the Sensor."""
+        super().__init__(coordinator)
+        self._platform_name = platform_name
+        self._site_id = site_id
+        self._data_service = data_service
+
+    @property
+    def name(self) -> str | None:
+        """Return the friendly name of the sensor."""
+        return self._platform_name + " - Price Spike"
+
+    @property
+    def unique_id(self) -> str | None:
+        """Return a unique id for each sensors."""
+        return slugify(self._site_id + " Price Spike")
+
+    @property
+    def native_value(self) -> bool:
+        """Return the current price in c/kWh."""
+        channel = self._data_service.current_prices.get(ChannelType.GENERAL)
+        if channel is not None:
+            return channel.spike_status == SpikeStatus.SPIKE
+        return False
+
+    @property
+    def icon(self):
+        """Return the icon of the sensor."""
+        channel = self._data_service.current_prices.get(ChannelType.GENERAL)
+        if channel is not None:
+            if channel.spike_status == SpikeStatus.SPIKE:
+                return "mdi:power-plug-off"
+            if channel.spike_status == SpikeStatus.POTENTIAL:
+                return "mdi:power-plug-outline"
+        return "mdi:power-plug"
+
+    @property
+    def device_state_attributes(self) -> Mapping[str, Any] | None:
+        """Return additional pieces of information about the sensor."""
+        data = {}
+        channel = self._data_service.current_prices.get(ChannelType.GENERAL)
+        if channel is not None:
+            data["spike_status"] = channel.spike_status.value
+        data[ATTR_ATTRIBUTION] = ATTRIBUTION
+        return data
+
+
+class AmberFactory:
+    """Create all the Amber sensors."""
+
+    def __init__(
+        self,
+        hass: HomeAssistant,
+        platform_name: str,
+        site_id: str,
+        api: amber_api.AmberApi,
+    ) -> None:
+        """Initialise the factory."""
+        self._platform_name = platform_name
+        self.data_service = AmberDataService(hass, api, site_id)
+        self._site_id = site_id
+
+    def build_sensors(self) -> list[SensorEntity]:
+        """Build and return all of the Amber Sensors."""
+        sensors: list[SensorEntity] = []
+
+        if (
+            self.data_service.site is not None
+            and self.data_service.coordinator is not None
+        ):
```

Comment:
This kind of checks, where you know that you won't do anything the rest of your function, turn them around.

```python
if bad_state:
    return sensors
```